src/test/java/pl/vm/javaguild/designpatterns/pattern/creational/prototype/FreeAgentMarketTest.java

```diff
@@ -0,0 +1,129 @@
+package pl.vm.javaguild.designpatterns.pattern.creational.prototype;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import pl.vm.javaguild.designpatterns.pattern.creational.prototype.adapter.FreeAgentMarketAdapter;
+import pl.vm.javaguild.designpatterns.pattern.creational.prototype.model.Coach;
+import pl.vm.javaguild.designpatterns.pattern.creational.prototype.model.CoachingStrategy;
+import pl.vm.javaguild.designpatterns.pattern.creational.prototype.model.Player;
+import pl.vm.javaguild.designpatterns.pattern.creational.prototype.model.SportDiscipline;
+import pl.vm.javaguild.designpatterns.pattern.creational.prototype.model.Team;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+import static org.assertj.core.api.Assertions.setAllowComparingPrivateFields;
+
+@DisplayName("Creational pattern: Prototype functionality Test")
+class FreeAgentMarketTest {
+
+    private final Player baseFirstPlayer = new Player("Michael Jordan", SportDiscipline.BASKETBALL);
+    private final Player baseSecondPlayer = new Player("Dennis Rodman", SportDiscipline.BASKETBALL);
+    private final Coach baseCoach = new Coach("Phil Jackson", CoachingStrategy.TRIANGLE);
+    private final Team baseTeam = new Team(baseCoach, List.of(baseFirstPlayer, baseSecondPlayer), "Chicago Bulls");
+
+    @Test
+    void should_create_player() {
+        // given
+        FreeAgentMarket freeAgentMarket = new FreeAgentMarketAdapter(baseFirstPlayer, baseCoach, baseTeam);
```

Comment:
imho testy to jest idealne miejsce aby uzywac var -> zatem zmienil bym kod w funkcjach testowych na np. final var freeAgentMarket = new FreeAgentMarketAdapter(baseFirstPlayer, baseCoach, baseTeam);